tests/test_default_gaussian.py

```diff
@@ -63,6 +63,21 @@ def prep_par(par, op):
         return [np.diag([x, 1]) for x in par]
     return par
 
+class TestExceptions(BaseTest):
+    """Tests that default.gaussian throws the correct error messages"""
+
+    def test_sample_exception(self):
+        """Test that default.gaussian raises an exception if sampling is attempted."""
+        self.logTestName()
+
+        dev = qml.device('default.gaussian', wires=1)
+
+        @qml.qnode(dev)
+        def circuit():
+            return qml.sample(qml.MeanPhoton(0), 10)
+        
+        with self.assertRaisesRegex(NotImplementedError, "Sampling is not supported in default.gaussian"):

```

Comment:
We could probably support it relatively easy for all ops except the fock state measurement. In all other cases, it's just a multivariate normal distribution that is sampled from. For the fock state measurement, we can keep unsupported and refer them to SF instead

Reply:
This can be done in a separate PR

Reply:
Good point. I'll open an issue.